Source/JobDriver_ZombieHunt.cs

```diff
@@ -10,29 +10,51 @@ namespace Zombiefied
     public class JobDriver_ZombieHunt : JobDriver
     {
         private int numMeleeAttacksMade;
-
-        public override void ExposeData()
-        {
-            base.ExposeData();
-            Scribe_Values.Look<int>(ref this.numMeleeAttacksMade, "numMeleeAttacksMade", 0, false);
-        }
-
         public override bool TryMakePreToilReservations(bool errorOnFailed)
         {
             return true;
         }
 
         protected override IEnumerable<Toil> MakeNewToils()
         {
-            yield return Toils_Combat.FollowAndMeleeAttack(TargetIndex.A, delegate
+            // If we can't reach the target, change the target to whatever's blocking us.
+            if(ZombiefiedMod.allowBreaching)
+            {
+                Toil gotoBlockers = GotoBlockers();
+                if (gotoBlockers != null) yield return gotoBlockers;
+            }
+            // Hit the target.
+            yield return HitThings();
+        }
+        public Toil GotoBlockers()
+        {
+            Thing thing;
+            using (PawnPath pawnPath = base.Map.pathFinder.FindPath(this.pawn.Position, base.TargetA.Cell, TraverseParms.For(this.pawn, Danger.Deadly, TraverseMode.PassAllDestroyableThings, false, false, false), PathEndMode.Touch, null))
+            {
+                if (!pawnPath.Found)
+                {
+                    this.EndJobWith(JobCondition.Incompletable);
+                    return null;
+                }
+                thing = pawnPath.FirstBlockingBuilding(out _, this.pawn);
+            }
+            if (thing != null)
+            {
+                this.job.targetA = thing;
+                return Toils_Goto.GotoThing(TargetIndex.A, PathEndMode.Touch);
+
+            }
+            return null;
+        }
+        public Toil HitThings()
+        {
+            this.job.killIncappedTarget = true;
+            this.job.maxNumMeleeAttacks = 5;
+            return Toils_Combat.FollowAndMeleeAttack(TargetIndex.A, delegate
             {
                 Thing thing = this.job.GetTarget(TargetIndex.A).Thing;
                 if (this.pawn.meleeVerbs.TryMeleeAttack(thing, this.job.verbToUse, false))
                 {
-                    if (this.pawn.CurJob == null || this.pawn.jobs.curDriver != this)
-                    {
-                        return;
-                    }
```

Comment:
Is there a reason to remove this

Reply:
oof, will revert, good catch, was learning things.